exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/MergeJoinBatch.java

```diff
@@ -255,84 +245,20 @@ private void setRecordCountInContainer() {
     }
   }
 
-  public void resetBatchBuilder() {
-    batchBuilder.close();
-    batchBuilder = new MergeJoinBatchBuilder(oContext.getAllocator(), status);
-  }
-
-  public void addRightToBatchBuilder() {
-    batchBuilder.add(right);
+  @Override
+  public void close() {
+    super.close();
+    leftIterator.close();
+    rightIterator.close();
   }
 
   @Override
   protected void killIncoming(boolean sendUpstream) {
+    close();
```

Comment:
You shouldn't call close here. I believe that should come from the FragmentExecutor.